Modules/posixmodule.c

```diff
@@ -12553,11 +12511,17 @@ DirEntry_from_find_data(path_t *path, WIN32_FIND_DATAW *dataW)
     }
 
     find_data_to_file_info(dataW, &file_info, &reparse_tag);
-    _Py_attribute_data_to_stat(&file_info, reparse_tag, &entry->win32_lstat);
+    if (!_Py_is_reparse_link(joined_path, reparse_tag, &is_link, TRUE)) {
+        PyErr_SetFromWindowsErr(GetLastError());
```

Comment:
Above we use `path_error` when this function failed, but here `GetLastError()`. Perhaps we want `PyErr_SetFromWindowsErrWithFilename()` in both places?

```suggestion
        PyErr_SetFromWindowsErrWithFilenameObject(PyExc_WindowsError, GetLastError(), entry->path);
```

Reply:
Your comment and code suggestion differ in which function to use. And the code suggestion does not exist. Looking at the definitions of the various error functions, I went with `PyErr_SetExcFromWindowsErrWithFilenameObject`.